Clarify parameter application formatting (#21978)
dotnet/docs: docs/fsharp/style-guide/formatting.md

## Formatting function parameter application

> In general, most function parameter application is done on the same line:

Reviewer: I'm not really sure this line makes sense - perhaps it's my TypeScript and C# background or unfamiliarity with F#, but the term "function parameter application" to me is just "argument", right?

```suggestion
In general, most arguments are provided on the same line:
```

```fsharp
let x = sprintf "\t%s - %i\n\r" x.IngredientName x.Quantity

let printListWithOffset a list1 =
    List.iter (fun elem -> printfn $"%d{a + elem}") list1
```

When pipelines are concerned, the same is typically also true, where a curried function is applied as an argument on the same line:

```
let printListWithOffsetPiped a list1 =
    list1
    |> List.iter (fun elem -> printfn $"%d{a + elem}")
```

> However, you may wish to apply parameters to a function on a new line, as a matter of readability or because the list of arguments or the argument names are too long. In that case, indent with one scope:

Reviewer: ```suggestion
However, you may wish to pass arguments to a function on a new line, as a matter of readability or because the list of arguments or the argument names are too long. In that case, indent with one scope:
```

Again, perhaps this is an F# term that is getting me a bit mixed up. But when I read "apply parameters" I think that I'm building out a function, which doesn't seem to be the case. Instead, it seems as though we're providing arguments to a function - thus, we "provide arguments" or "pass arguments" - not "apply parameters, right?

Author: It's technically more correct to call it parameter application, but I think this is fine for an explanation
